src/raw/cgroup.go

```diff
@@ -270,6 +277,7 @@ func memory(metric *cgroups.Metrics) (Memory, error) {
 	mem.Cache = metric.Memory.Cache
 	mem.RSS = metric.Memory.RSS
 	mem.SwapUsage = metric.Memory.Swap.Usage
-
+	mem.SwapLimit = metric.Memory.HierarchicalSwapLimit
```

Comment:
I think we are not being consistent with the use of `HierarchicalSwapLimit`. For getting the memory limit we use `Memory.Usage.Limit` rather than `HierarchicalMemoryLimit`, so we should just pick one of the two approaches and go with it.

Regarding which of the two is more correct, I'm not entirely sure. From my understanding, hierarchical limits are not enabled by default in containerd (not sure about Docker), and while one could use them to, for example, limit the amount of memory that a group of containers (or all containers in the system) can claim, I do not see this as common usage.

Reply:
good catch thanks! yes this is inconstant, I will use  `metric.Memory.Swap.Limit` then herarchical can be added if needed with explicit reference in the name